cmake/toolchains/gcc_gvsoc_pulpissimo.cmake

```diff
@@ -10,11 +10,38 @@ SET(CMAKE_LINKER ${TC_PREFIX}ld${EXE_EXT})
 SET(CMAKE_OBJCOPY ${TC_PREFIX}objcopy${EXE_EXT})
 SET(CMAKE_AR ${TC_PREFIX}ar${EXE_EXT})
 SET(CMAKE_RANLIB ${TC_PREFIX}ranlib${EXE_EXT})
-
 # The linker argument setting below will break the cmake test program on 64-bit, so disable test program linking for
 # now.
 SET(CMAKE_TRY_COMPILE_TARGET_TYPE "STATIC_LIBRARY")
 
-SET(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -msmall-data-limit=8 -mno-save-restore -fsigned-char -g3 -D__PULP_USE_LIBC -DSTDIO_FAKE=2 -DSTDIO_UART=1 -DSTDIO_NULL=0 -DCONFIG_STDIO=2 -DportasmHANDLE_INTERRUPT=undefined_handler")
-SET(CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} -msmall-data-limit=8 -mno-save-restore -fsigned-char -std=gnu11 -Wall -Wextra -Wshadow -Wformat=2 -Wundef -Wsign-conversion -Wno-unused-parameter -g3 -D__PULP_USE_LIBC -DSTDIO_FAKE=2 -DSTDIO_UART=1 -DSTDIO_NULL=0 -DCONFIG_STDIO=2")
-SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -msmall-data-limit=8 -mno-save-restore -fsigned-char -std=gnu11 -Wall -Wextra -Wshadow -Wformat=2 -Wundef -Wsign-conversion -Wno-unused-parameter -g3 -nostartfiles -Wl,--gc-sections -Wl,-Map,memory.map")
\ No newline at end of file
+# the following is transfered from PULP-FreeRTOS repository
+# https://github.com/pulp-platform/pulp-freertos/blob/master/default_flags.mk#L110-L126
+# Builtin mandatory flags. Need to be simply expanded variables for appends in
+# other cmake files to work correctly 
+SET(CV_CFLAGS "\
+-msmall-data-limit=8 -mno-save-restore \
+-fsigned-char -ffunction-sections -fdata-sections \
+-std=gnu11 \
+-Wall -Wextra -Wshadow -Wformat=2 -Wundef -Wsign-conversion -Wno-unused-parameter")
+
+SET(CV_ASFLAGS "\
+-msmall-data-limit=8 -mno-save-restore \
+-fsigned-char -ffunction-sections -fdata-sections \
+-x assembler-with-cpp")
+
+SET(CV_CPPFLAGS)
+
+# note: linkerscript is included in target directory makefile.mk
+SET(CV_LDFLAGS "\
+-nostartfiles \
+-Wl,--gc-sections \
+-Wl,-Map,memory.map") # -Wl,--print-gc-sections
+
+# LIST(APPEND CV_LDFLAGS " -Wstack-usage=1024")
+
+# the following is transfered from PULP-FreeRTOS repository
+# https://github.com/pulp-platform/pulp-freertos/blob/master/default_targets.mk#L77-L87
+SET(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} ${CV_CFLAGS} ${CV_CPPFLAGS} -DSTDIO_FAKE=2 -DSTDIO_UART=1 -DSTDIO_NULL=0 -DCONFIG_STDIO=2 -Og ")
```

Comment:
@zhouxinyu0723 What are those `-Og` for at the end of each line?

Reply:
It means optimizing code for debugging. This should be removed and I have pushed a new commit removing them.